packages/blocks/src/list-block/list-service.ts

```diff
@@ -2,24 +2,28 @@ import { BlockService } from '@blocksuite/block-std';
 
 import { InlineManager } from '../_common/inline/inline-manager.js';
 import {
-  affineInlineSpecs,
   type AffineTextAttributes,
+  getAffineInlineSpecsWithReference,
 } from '../_common/inline/presets/affine-inline-specs.js';
 import { affineInlineMarkdownMatches } from '../_common/inline/presets/markdown.js';
+import { ReferenceNodeConfig } from '../_common/inline/presets/nodes/reference-node/reference-config.js';
 import type { ListBlockModel } from './list-model.js';
 
 export class ListService<
   TextAttributes extends AffineTextAttributes = AffineTextAttributes,
 > extends BlockService<ListBlockModel> {
   readonly inlineManager = new InlineManager<TextAttributes>();
+  readonly referenceNodeConfig = new ReferenceNodeConfig();
 
   override mounted(): void {
     super.mounted();
-    (
-      this.inlineManager as unknown as InlineManager<AffineTextAttributes>
-    ).registerSpecs(affineInlineSpecs);
-    (
-      this.inlineManager as unknown as InlineManager<AffineTextAttributes>
-    ).registerMarkdownMatches(affineInlineMarkdownMatches);
+
+    this.referenceNodeConfig.setPage(this.page);
```

Comment:
The page reference may change.

Reply:
What do you mean? When a new page is loaded, the mount method will be re-executed, and the config will be updated.

Reply:
Good to know!